homeassistant/components/nina/config_flow.py

```diff
@@ -0,0 +1,138 @@
+"""Config flow for Nina integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from pynina import ApiError, Nina
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    _LOGGER,
+    CONF_FILTER_CORONA,
+    CONF_MESSAGE_SLOTS,
+    CONF_REGIONS,
+    CONST_REGION_MAPPING,
+    CONST_REGIONS,
+    DOMAIN,
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for NINA."""
+
+    VERSION: int = 1
+
+    def __init__(self) -> None:
+        """Initialize."""
+        super().__init__()
+        self._all_region_codes_sorted: dict[str, str] = {}
+        self.regions: dict[str, dict[str, Any]] = {}
+
+        for name in CONST_REGIONS:
+            self.regions[name] = {}
+
+    async def async_step_user(
+        self: ConfigFlow,
+        user_input: dict[str, Any] | None = None,
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, Any] = {}
+
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        has_error: bool = False
+
+        if len(self._all_region_codes_sorted) == 0:
+            try:
+                nina: Nina = Nina()
```

Comment:
Can instantiating the Nina class raise? Otherwise please move it out of the try... except block.